sdk/core/Azure.Core/tests/HttpPipelineFunctionalTests.cs

```diff
@@ -158,6 +158,82 @@ public async Task NonBufferedFailedResponsesAreDisposedOf()
             Assert.Greater(reqNum, requestCount);
         }
 
+        [Test]
+        public async Task Opens50ParallelConnections()
+        {
+            // Running 50 sync requests on the threadpool would cause starvation
+            // and the test would take 20 sec to finish otherwise
```

Comment:
Are we doing anything to validate that we're not hitting starvation?

Reply:
Not really, this is just to speedup tests. Starvation is expected here because threadpool is smaller than 50 by-default and we are completely blocking threads.